dashboard/test/models/section_test.rb

```diff
@@ -66,4 +66,58 @@ class SectionTest < ActiveSupport::TestCase
 
     assert !Section.exists?(section.id)
   end
+
+  test 'add_student adds student to section' do
+    section = create :section
+    student = create :student
+
+    # even when the student is in another section from a different teacher
+    create(:section).add_student student
+
+    assert_creates(Follower) do
+      section.add_student student
+    end
+    assert section.students.exists?(student.id)
+    assert_equal 2, Follower.count(student_user_id: student.id)
+  end
+
+  test 'add_student in another section from the same teacher moves student' do
+    teacher = create :teacher
+    student = create :student
+    original_section = create :section, user: teacher
+    original_section.add_student student
+    new_section = create :section, user: teacher
+
+    # Initially, student is in original_section
+    assert original_section.students.exists?(student.id)
+    refute new_section.students.exists?(student.id)
+
+    assert_does_not_create(Follower) do
+      new_section.add_student student
+    end
+
+    # Verify student has been moved to new_section
+    refute original_section.students.exists?(student.id)
+    assert new_section.students.exists?(student.id)
+  end
+
+  test 'add_student with move_for_same_teacher: false does not move student' do
```

Comment:
I wonder if this is confusing for anyone else. Terminology wise we tend to use student and follower interchangeably (same with teacher and section owner). In this test we're confirming the follower stays in both sections, but in reality we never want this to be true for _real students_ right? Not something to block this PR, just some thoughts.